### PR TITLE
Refactor: Move connectivity check to http

### PR DESCRIPTION
**Pull request by DanVanAtta**
Migrate connectivity test to http server from java socket server.
The connectivity check is the verification that kicks in after hosting
a game where the server tries to establish a 'reverse' connection
to the host to make sure the host is accessible from the internet.

This update moves that check to HTTP server.

 

## Functional Changes

[ ] New map or map update
[ ] New Feature
[ ] Feature update or enhancement
[ ] Feature Removal
[x] Code Cleanup or refactor
[ ] Configuration Change
[ ] Bug fix:  
[ ] Other:   

## Testing

[ ] No manual testing done
[x] Manually tested

- launched a local lobby with bot
- joined local lobby, hosted a game

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/ServerSetupPanel.java [hunk 1]
```diff
@@ -100,11 +105,24 @@ private void createLobbyWatcher() {
...
+                  .localPort(model.getMessenger().getLocalNode().getPort())
                   .errorHandler(errorHandler)
-                  .localNode(model.getMessenger().getLocalNode())
                   .build()
                   .run();
+
+              System.clearProperty(LOBBY_HOST);
+              System.clearProperty(LOBBY_PORT);
+              System.clearProperty(LOBBY_HTTPS_PORT);
+              System.clearProperty(TRIPLEA_NAME);
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/startup/ui/ServerSetupPanel.java]: Could you explain why this change was introduced?

File: game-core/src/test/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcherTest.java [hunk 2]
```diff
@@ -46,15 +46,6 @@ void shouldReturnPrimaryValueWhenPrimaryValueSet() {
...
     }
 
-    @Test
-    void shouldClearPrimaryValueWhenPrimaryValueSet() {
-      givenPrimaryValueSet();
-
-      getLobbySystemProperty(KEY);
-
-      assertThat(System.getProperty(KEY), is(nullValue()));
-    }
```
Comment (RoiEXLab) [inline game-core/src/test/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcherTest.java]: What about reversing this test now that the clear statement is removed?

File: http-server/src/main/java/org/triplea/server/lobby/game/ConnectivityController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  @RateLimited(
+      keys = {KeyPart.IP},
+      rates = {@Rate(limit = 10, duration = 1, timeUnit = TimeUnit.MINUTES)})
+  @RolesAllowed(UserRole.ANONYMOUS)
+  public boolean checkConnectivity(@Context final HttpServletRequest request, final Integer port) {
+    Preconditions.checkArgument(port > 0, "Port must be a positive number, was: " + port);
+    Preconditions.checkArgument(
+        port < Math.pow(2, 16), "Port must be less than max value (2^16), was: " + port);
+
+    return connectivityCheck.test(new InetSocketAddress(request.getRemoteAddr(), port));
```
Comment (RoiEXLab) [inline http-server/src/main/java/org/triplea/server/lobby/game/ConnectivityController.java]: I wonder what was your motivation was to move this check to the server?

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/LocalServerAvailabilityCheck.java [hunk 4]
```diff
@@ -12,15 +11,20 @@
...
+  /**
+   * Starts a thread that requests the server to do a 'reverse' connection back to the local client.
+   * This is to ensure that our local IP address is public facing and that other computers on the
+   * internet can establish connections to this (the local) host.
+   */
   public void run() {
     // if we lose our connection, then shutdown
     new Thread(
             () -> {
-              if (!controller.testGame(localNode)) {
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/framework/startup/ui/LocalServerAvailabilityCheck.java]: Hmm I'm not seeing that this method is removed, are there any usages?

Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/framework/startup/ui/LocalServerAvailabilityCheck.java]: Casualty of working on this from multiple machines, change was made but did not make it into this branch. Now removed: 949f6e9b8bd32f243cd5c89c839186c5f335cb12

File: game-core/src/main/java/games/strategy/engine/framework/startup/ui/ServerSetupPanel.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/framework/startup/ui/ServerSetupPanel.java]: Turns out INode.getAddress().getHostAddress()... etc.. returns the interface in addition to IP address. Hence this warning https://github.com/triplea-game/triplea/pull/5304/files#diff-5c823a8b563c30a1fc7db84682121912R26

That can't be used to create a valid URI, and we need to create one for the HttpLobbyClient. We do have the host address available, but lobby watcher has the side effect of clearing it (so that way on the next game launch the magic system property does not cause the game to launch in a hosting mode). But, we do need lobby watcher as it creates a connection which gives us an api key. So.. the clear is removed from lobby watcher and moved to after we create the http client so that we can get the host address from props and also get the API key.

File: game-core/src/test/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcherTest.java (hunk 2 above)
Comment (DanVanAtta) [reply game-core/src/test/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcherTest.java]: It's okay to delete tests sometimes :smile: 
On another level, it's pretty special that the lobby watcher cleared a system property, I don't think we need to be explicit about it being left alone. Ideally we would move this check to code that is testing where we do now have the property clear, though that is a bit prohibitive/not worth it.

File: http-server/src/main/java/org/triplea/server/lobby/game/ConnectivityController.java (hunk 3 above)
Comment (DanVanAtta) [reply http-server/src/main/java/org/triplea/server/lobby/game/ConnectivityController.java]: Removes a dependency on messengers, eventually with the goal of having no java NIO lobby left.
